File: app/src/at-hungtran/java/asiantech/internship/summer/restapi/APIImages.java
package asiantech.internship.summer.restapi;

import java.util.List;

import asiantech.internship.summer.model.ImageItem;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Query;
import retrofit2.http.Url;

public interface APIImages {
    String TOKEN = "6f5a48ac0e8aca77e0e8ef42e88962852b6ffaba01c16c5ba37ea13760c0317e";
    String BASE_URL = "https://api.gyazo.com/api/";
    String UPLOAD_URL = "https://upload.gyazo.com/api/upload";

    @GET("images")
    Call<List<ImageItem>> getImages(@Query("access_token") String token,
                                    @Query("page") int page);

    @Multipart
    @POST
    Call<ImageItem> uploadImage(@Url String url, @Part("access_token") RequestBody token, @Part MultipartBody.Part image);
}

File: app/src/at-hungtran/java/asiantech/internship/summer/kotlin/restapi/APIImages.kt
package asiantech.internship.summer.kotlin.restapi

import asiantech.internship.summer.kotlin.model.ImageItem
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.*

interface APIImages {

    @GET("images")
    fun getImages(@Query("access_token") token: String,
                           @Query("page") page: Int): Call<List<ImageItem>>

    @Multipart
    @POST
    fun uploadImage(@Url url: String, @Part("access_token") token: RequestBody, @Part image: MultipartBody.Part): Call<ImageItem>

    companion object {
        const val TOKEN = "604d1f2a63e1620f8e496970f675f0322671a3de0ba9f44c850e9ddc193f4476"
        const val BASE_URL = "https://api.gyazo.com/api/"
        const val UPLOAD_URL = "https://upload.gyazo.com/api/upload"
    }

}